cannon/mipsevm/memory/mpt.go

```diff
@@ -0,0 +1,485 @@
+//go:build cannon64
+// +build cannon64
+
+package memory
+
+import (
+	"math/bits"
+)
+
+// Representation of the state of the memory as a Merkle Patricia Trie (MPT). Only for 64 bit Cannon.
+//
+// This implementation is copied from asterisc.
+type TrieIndex struct {
```

Comment:
Given we didn't see a significant performance improvement from this version, it's probably better for cannon to stick with the binary tree PageIndex implementation and just remove this one so we have some differentiation between asterisc and cannon.

I think the `PageIndex` refactor is still worthwhile and improves maintainability even if we only ever have one implementation, but it will also make it easier for potential future experimentation with other implementations.